cmd/tmpfs/tmpfs.go

```diff
@@ -0,0 +1,347 @@
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"sync"
+
+	"github.com/Harvey-OS/go/internal/tmpfs"
+	"github.com/Harvey-OS/ninep/pkg/debugfs"
+	"github.com/Harvey-OS/ninep/protocol"
+)
+
+var (
+	networktype = flag.String("ntype", "tcp4", "Default network type")
+	netaddr     = flag.String("addr", ":5641", "Network address")
+	debug       = flag.Int("debug", 0, "Print debug messages")
+	pkg         = flag.String("package", "", "tar.gz package to open")
+)
+
+type fileServer struct {
+	Versioned bool
+
+	archive    *tmpfs.Archive
+	filesMutex sync.Mutex
+	files      map[protocol.FID]*FidEntry
+	ioUnit     protocol.MaxSize
+}
+
+// FidEntry wraps an Entry with the instance data required for a fid reference
+type FidEntry struct {
+	tmpfs.Entry
+
+	// Username to be used for all entries in this hierarchy
+	uname string
+
+	// We can't know how big a serialized dentry is until we serialize it.
+	// At that point it might be too big. We save it here if that happens,
+	// and on the next directory read we start with that.
+	oflow []byte
+
+	// Index of next child to return when reading a directory
+	nextChildIdx int
+}
+
+func newFidEntry(entry tmpfs.Entry, uname string) *FidEntry {
+	return &FidEntry{Entry: entry, uname: uname, oflow: nil, nextChildIdx: -1}
+}
+
+func (fs *fileServer) Rversion(msize protocol.MaxSize, version string) (protocol.MaxSize, string, error) {
```

Comment:
you should run 
go vet
goling

and you'll get complaints, go ahead and fix them.

Reply:
what do you mean by 'goling'?

Reply:
I ran go vet and had no complaints.  What did you think it would find?

Reply:
ah hell I meant
golint

it will complain about every function that starts with a capital letter and has no doc comment.

Reply:
ok that makes more sense, will look